hazelcast/src/main/java/com/hazelcast/query/impl/UnorderedIndexStore.java

```diff
@@ -136,6 +140,44 @@ public boolean canEvaluate(Class<? extends Predicate> predicateClass) {
         throw new UnsupportedOperationException();
     }
 
+    @Override
+    public Iterator<QueryableEntry> getRecordIterator() {
+        Iterator<QueryableEntry> iterator = new IndexEntryCompositeIterator(recordMap.values().iterator());
+        Iterator<QueryableEntry> nullIterator = recordsWithNullValue.values().iterator();
+
+        return new FlatCompositeIterator<>(Arrays.asList(iterator, nullIterator).iterator());
+    }
+
+    @Override
+    public Iterator<QueryableEntry> getRecordIterator(Comparable value) {
+        if (value == NULL) {
+            return recordsWithNullValue.values().iterator();
+        } else {
+            Map<Data, QueryableEntry> res = recordMap.get(canonicalize(value));
+
+            if (res == null) {
+                return Collections.emptyIterator();
+            }
+
+            return res.values().iterator();
+        }
+    }
+
+    @Override
+    public Iterator<QueryableEntry> getRecordIterator(Comparison comparison, Comparable value) {
+        throw new UnsupportedOperationException();
```

Comment:
In the old `Set` based implementation we supported a similar method. Yes, not efficiently, but we didn't throw an exception.

Reply:
We do not use HASH indexes for range lookups - it doesn't make sense from the performance perspective. Therefore, this method is not implemented - it is just never called.

Reply:
@taburet , for the Set -based methods we do have an implementation of similar methods. Can we still use them?

Reply:
Yes, there are implementations for Set-based methods. The key question - why do we want to use them at all? 

Currently, the planner do not consider HASH indexes for range requests. And therefore these unsupported methods are never invoked. The motivation for this is that the HASH index cannot speedup range queries.

Reply:
@petrpleshachkov Yes, the old set-based methods are still used in the old engine. To my understanding the idea of using hash indexes for range full scans was to avoid deserialization/extraction of the attribute in question since index keys are already extracted/deserialized. Never measured the actual performance benefits, but there should be some :)